manifests/database/postgresql.pp

```diff
@@ -26,33 +32,39 @@
     # Temporary direct use of liquibase to initiall migrate the candlepin database
     # until support is added in cpdb - https://bugzilla.redhat.com/show_bug.cgi?id=1044574
     include ::postgresql::client, ::postgresql::server
-    postgresql::server::db { $candlepin::db_name:
-      user     => $candlepin::db_user,
-      password => postgresql_password($candlepin::db_user, $candlepin::db_password),
+    postgresql::server::db { $db_name:
+      user     => $db_user,
+      password => postgresql_password($db_user, $db_password),
       encoding => 'utf8',
       locale   => 'en_US.utf8',
-    } ~>
+    }
+    Postgresql::Server::Role[$db_user] -> Postgresql::Server::Database[$db_name]
+  }
+
+  if $init_db {
     exec { 'cpdb':
       path        => '/bin:/usr/bin',
       command     => "liquibase --driver=org.postgresql.Driver \
-                            --classpath=/usr/share/java/postgresql-jdbc.jar:/var/lib/${candlepin::tomcat}/webapps/candlepin/WEB-INF/classes/ \
+                            --classpath=/usr/share/java/postgresql-jdbc.jar:/var/lib/${tomcat}/webapps/candlepin/WEB-INF/classes/ \
                             --changeLogFile=db/changelog/changelog-create.xml \
-                            --url=jdbc:postgresql:${candlepin::db_name} \
-                            --username=${candlepin::db_user}  \
-                            --password=${candlepin::db_password} \
+                            --url=jdbc:postgresql:${db_host}:${db_port}/${db_name} \
+                            --username=${db_user}  \
+                            --password=${db_password} \
                             migrate \
                             -Dcommunity=False \
-                            >> ${candlepin::log_dir}/cpdb.log \
+                            >> ${log_dir}/cpdb.log \
                             2>&1 && touch /var/lib/candlepin/cpdb_done",
-      creates     => "${candlepin::log_dir}/cpdb_done",
+      creates     => '/var/lib/candlepin/cpdb_done',
       refreshonly => true,
-      before      => Service[$candlepin::tomcat],
+      before      => Service[$tomcat],
```

Comment:
I'm pretty sure its redundant, but I wanted to keep my code changes to a minimum; I'll clean it up